File: Java/src/com/dbc/algorithm/ReferOffer/LianBiaoZhongDaoShuDiKgeJieDianLcof.java
package com.dbc.algorithm.ReferOffer;

public class LianBiaoZhongDaoShuDiKgeJieDianLcof {
    public class ListNode {
        int val;
        ListNode next;

        ListNode(int x) {
            val = x;
        }
    }

    public ListNode getKthFromEnd(ListNode head, int k) {
        ListNode front = head, back = head;
        for (int i = 0; i < k - 1; i++) {
            if (back == null) {
                return head;
            }
            back = back.next;
        }
        while (back.next != null) {
            front = front.next;
            back = back.next;
        }
        return front;
    }
}

File: src/tmp/Python/refer-offer/lian_biao_zhong_dao_shu_di_kge_jie_dian_lcof.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def getKthFromEnd(head: ListNode, k: int) -> ListNode:
    front, back = head, head
    for i in range(k - 1):
        if not back:
            return head
        back = back.next
    
    while back.next:
        front = front.next
        back = back.next
    
    return front